documentation/src/main/doc/modules/kafka/examples/inbound/KafkaRebalancedConsumerRebalanceListener.java

```diff
@@ -0,0 +1,86 @@
+package inbound;
+
+import io.smallrye.reactive.messaging.kafka.IncomingKafkaRecord;
+import io.smallrye.reactive.messaging.kafka.KafkaConsumerRebalanceListener;
+import io.vertx.kafka.client.common.TopicPartition;
+import io.vertx.mutiny.kafka.client.consumer.KafkaConsumer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+@Named("rebalanced-example.rebalancer")
+public class KafkaRebalancedConsumerRebalanceListener implements KafkaConsumerRebalanceListener {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaRebalancedConsumerRebalanceListener.class.getName());
+
+    /**
+     * With this mapping we can keep track of which partitions have been set-up properly.
+     */
+    private final List<Long> offsetMap = new CopyOnWriteArrayList<>();
+
+    /**
+     * When receiving a list of partitions will search for the earliest offset within 10 minutes
+     * and seek the consumer to it. These operations are asynchronous so the inbound connector
+     * WILL continue to receive messages from the subscribed topic that MIGHT be older than 10 minutes.
+     *
+     * @param consumer underlying consumer
+     * @param topicPartitions set of assigned topic partitions
+     */
+    @Override
+    public void onPartitionsAssigned(KafkaConsumer<?, ?> consumer, Set<TopicPartition> topicPartitions) {
```

Comment:
Instead of "void" should it returns a `Uni<Void>` indicating when the action has completed?

Reply:
done